### ISSUE TITLE
Workspaces is unusable

### ISSUE THREAD

**Issue by mroberts1** [notaapp/issues#41]
Nota is, by some distance, the most beautifully-designed markdown editor I have tried over the past six months. Purchasing a license was a no-brainer. So why have I been unable to take to it?

The problem is very simple: the workspace. When launched, the app requires you to set a home directory (aka workspace), whether locally or on a service like Dropbox. This choice immediately put me into a quandary, since I was not yet sure where I wanted to put my workspace: my root directory? iCloud? Dropbox? I wasn't sure, but I had to decide. I decided, for the moment, on iCloud, assuming that I could change it later.

I couldn't. Not only is it not possible to have multiple workspaces (at least if that feature exists, I couldn't figure out how to access it); apparently, it isn't even possible to change the existing  workspace to a different directory (again, unless I've missed something). 

If you want to change the home directory, you basically have to reinstall the app and make a different selection at the initial choice window. 

After doing this several times, I have started to lose track of where my home directory is - my home folder shows simply "root" but I have no idea where that actually is anymore. 

The Go menu has a "Go to workspace" option that does not open the local (or online) folder the way (say) GitHub Desktop does, and does not, it seems, do anything at all.

The confusion over this trivial issue is an unnecessary distraction in what is supposed to be a "distractionless" app. It is very disappointing given the great care that seems to have gone into everything else about the app. 

I'm sorry to post such a negative comment, but this is the single issue why Nota is currently not my default markdown editor, and it will unfortunately stay that way until the issue is addressed.

**Comment by astoilkov** [comment]
`File -> Open...` can open a new workspace. Dragging and dropping a folder onto the Nota editor also works.

> The confusion over this trivial issue is an unnecessary distraction in what is supposed to be a "distractionless" app. It is very disappointing given the great care that seems to have gone into everything else about the app.

I agree. Sorry for the confusion. You are actually not the first person to be confused by this: https://github.com/notaapp/issues/issues/34#issuecomment-748134007. This is why — for the next release — we will be adding a `File -> Open Workspace...` item in the menus.

> The Go menu has a "Go to workspace" option that does not open the local (or online) folder the way (say) GitHub Desktop does, and does not, it seems, do anything at all.

Every time you open a new workspace using `File -> Open...` that workspace will be available in the `Go -> Go to Workspace...` for quick access.

Quoting from the additional email you send us:
> I was able to locate where my "root" directory is (iCloud) by right-clicking on it in the app. But I then discovered that only the first 10 files in that directory are listed in the Folder > Files view in Nota. So that appears to be another problem.

We only show text, markdown, and image files in the Folder > Files view in Nota. This can be a reason you are missing some of your files.

Additionally, I am not sure if I understood what you mean by "I was able to locate where my "root" directory is". Isn't it in the list of folders inside the Folders view?

P.S. You also send us an email. If you don't mind, we can continue the conversation here so it is useful for others as well.

**Comment by mroberts1** [comment]
Thanks for these helpful comments - I didn't realize that File > Open was used to switch between workspaces, so renaming it as File > Open Workspace would be helpful (maybe separately from actual File > Open?)

I think the problem arises (as it does with a lot of Markdown-based tools like Obsidian) from the assumption that people will just keep ALL of their markdown files in the same main directory. This is certainly best practice, but reality - at least my reality - is more messy in that they're scattered all over the place. So being able to toggle easily between multiple workspaces is a useful feature, even if it may not be as good as keeping everything in the same folder.

This is nevertheless a reminder that I need to clean up my drive. ;-)

**Comment by astoilkov** [comment]
> Thanks for these helpful comments - I didn't realize that File > Open was used to switch between workspaces, so renaming it as File > Open Workspace would be helpful (maybe separately from actual File > Open?)

Yes, we will have both `File -> Open...` and `File -> Open Workspace...`

> I think the problem arises (as it does with a lot of Markdown-based tools like Obsidian) from the assumption that people will just keep ALL of their markdown files in the same main directory. This is certainly best practice, but reality - at least my reality - is more messy in that they're scattered all over the place. So being able to toggle easily between multiple workspaces is a useful feature, even if it may not be as good as keeping everything in the same folder.

Just a suggestion: you can hide the sidebars and just drag & drop files onto Nota. After the first open you can access the file through the `Go -> Go to File...` faster.

**Comment by astoilkov** [comment]
In the last few updates, we made quite a few improvements to make workspaces easier to understand. It should be easier for people to understand the concept and get started.

- We moved the Switch Workspace item inside the File menu.
- We wrote a text explaining how to switch workspace on the welcome screen.
- We made the workspace popup move visually appealing.
- We made the workspace popup have an item that gives you an option to open a folder anywhere on your computer.

@mroberts1 What do you think about these changes?

**Comment by astoilkov** [comment]
Hi @lolbat,

We have been talking with @erusev about this and we think your suggestions make sense. We have the same problem sometimes. We will try to figure out an elegant solution to this.

Thanks for your feedback!
